src/node.cc

```diff
@@ -1436,7 +1436,16 @@ void FatalException(Isolate* isolate,
       exit(7);
     } else if (caught->IsFalse()) {
       ReportException(env, error, message);
-      exit(1);
+
+      // fatal_exception_function call before may have set a new exit code ->
+      // read it again, otherwise use default for uncaughtException 1
+      Local<String> exit_code = env->exit_code_string();
+      MaybeLocal<Value> code = process_object->Get(env->context(), exit_code);
+      if (code.IsEmpty()) {
+        exit(1);
+      } else {
+        exit(code.ToLocalChecked()->Int32Value(env->context()).ToChecked());
+      }
```

Comment:
This crashes for weird edge cases like this:

```js
process.on("exit", () => {
  process.exitCode = { [Symbol.toPrimitive]() { throw new Error(); } };
})
```

Maybe exit with 1 if `code.IsEmpty() || !code->IsInt32()` and use a direction conversion to `v8::Int32` here, like this:

https://github.com/nodejs/node/blob/1f1675817c6ed7965f3e5af173ce9f736d3cced0/src/node_api.cc#L2134

Does that sound okay?

Reply:
Wow, that's pretty neat. Though, one question, is it safe to just ` Local<Value> code = process_object->Get(env->context(), exit_code).ToLocalChecked();`? Or should I do it step by step (empty check, then convert and int32 check).

Reply:
@lundibundi That operation can still fail if userland code were to install a getter that throws … obviously not something it *should* do, but yes, I’d prefer to do it step-by-step.

(Generally, we’ll need to shift a lot of our error handling code away from using `ToLocalChecked()`/`ToChecked()`, because `worker.terminate()` can make just about anything throw that calls JS code…)

Reply:
Ok, thanks, I'll update it soon.